apps/bttester/src/btp_pacs.c

```diff
@@ -0,0 +1,331 @@
+    #ifdef MYNEWT_VAL_BLE_SVC_AUDIO_PACS_LC3_SRC_MAX_CODEC_FRAMES_PER_SDU
+        MYNEWT_VAL(BLE_SVC_AUDIO_PACS_LC3_SNK_MAX_CODEC_FRAMES_PER_SDU),
+    #endif
+);
+
+static uint8_t ble_svc_audio_pacs_lc3_src_codec_spec_caps[] = BLE_AUDIO_BUILD_CODEC_CAPS(
+    MYNEWT_VAL(BLE_SVC_AUDIO_PACS_LC3_SRC_SAMPLING_FREQUENCIES),
+    MYNEWT_VAL(BLE_SVC_AUDIO_PACS_LC3_SRC_FRAME_DURATIONS),
+#ifdef MYNEWT_VAL_BLE_SVC_AUDIO_PACS_LC3_SRC_AUDIO_CHANNEL_COUNTS
+    MYNEWT_VAL(BLE_SVC_AUDIO_PACS_LC3_SRC_AUDIO_CHANNEL_COUNTS),
+#else
+    ,
+#endif
+    MYNEWT_VAL(BLE_SVC_AUDIO_PACS_LC3_SRC_MIN_OCTETS_PER_CODEC_FRAME),
+    MYNEWT_VAL(BLE_SVC_AUDIO_PACS_LC3_SRC_MAX_OCTETS_PER_CODEC_FRAME),
+#ifdef MYNEWT_VAL_BLE_SVC_AUDIO_PACS_LC3_SRC_MAX_CODEC_FRAMES_PER_SDU
+    MYNEWT_VAL(BLE_SVC_AUDIO_PACS_LC3_SRC_MAX_CODEC_FRAMES_PER_SDU),
+#endif
+);
+
+static struct ble_audio_codec_register_params snk_codec_params = {
+    .codec_id = {
+        .format = BLE_SVC_AUDIO_PACS_LC3_CODEC_ID,
+        .company_id = 0x00,
+        .vendor_specific = 0x00
+    },
+    .codec_spec_caps_len = sizeof(ble_svc_audio_pacs_lc3_snk_codec_spec_caps),
+    .codec_spec_caps = ble_svc_audio_pacs_lc3_snk_codec_spec_caps,
+#ifdef MYNEWT_VAL_BLE_SVC_AUDIO_PACS_LC3_SNK_METADATA
+    .metadata_len = sizeof(ble_svc_audio_pacs_lc3_snk_metadata),
+    .metadata = ble_svc_audio_pacs_lc3_snk_metadata,
+#else
+    .metadata_len = 0,
+#endif
+
+    .direction = BLE_AUDIO_CODEC_DIR_SINK_BIT
+};
+
+static struct ble_audio_codec_register_params src_codec_params = {
+    .codec_id = {
+        .format = BLE_SVC_AUDIO_PACS_LC3_CODEC_ID,
+        .company_id = 0x00,
+        .vendor_specific = 0x00
+    },
+    .codec_spec_caps_len = sizeof(ble_svc_audio_pacs_lc3_src_codec_spec_caps),
+    .codec_spec_caps = ble_svc_audio_pacs_lc3_src_codec_spec_caps,
+#ifdef MYNEWT_VAL_BLE_SVC_AUDIO_PACS_LC3_SRC_METADATA
+    .metadata_len = sizeof(ble_svc_audio_pacs_lc3_src_metadata),
+    .metadata = ble_svc_audio_pacs_lc3_src_metadata,
+#else
+    .metadata_len = 0,
+#endif
+    .direction = BLE_AUDIO_CODEC_DIR_SOURCE_BIT
+};
+
+int
+set_available(uint16_t conn_handle, void *arg)
+{
+    int rc;
+    struct set_avail_cb_data *avail_data = arg;
+
+    rc = ble_svc_audio_pacs_avail_contexts_set(conn_handle,
+                                               avail_data->snk_ctxts,
+                                               avail_data->src_ctxts);
+    if (rc) {
+        return BTP_STATUS_FAILED;
+    }
+
+    return BTP_STATUS_SUCCESS;
+}
+
+static uint8_t
+pacs_set_available_contexts(const void *cmd, uint16_t cmd_len,
+                            void *rsp, uint16_t *rsp_len)
+{
+    struct set_avail_cb_data cb_data;
+    const struct pacs_set_available_contexts_cmd *avail = cmd;
+
+    /* If this originated from pacs_update_characteristic - we update with unspecified */
+    if (avail->sink_contexts == BTP_PACS_CHARACTERISTIC_AVAILABLE_AUDIO_CONTEXTS) {
```

Comment:
those are le16,  should be converted to host order before read

Reply:
Done